### PR TITLE
Chore: Add [in|out-of] cluster examples and enhance docs

### PR DESCRIPTION
**Pull request by yue9944882**
/cc @brendandburns

### CODE REVIEW

**Comment by yue9944882** [comment]
Fixes: #621

File: README.md [hunk 1]
```diff
@@ -61,70 +61,46 @@ Then manually install the following JARs:
...
-                api.listNamespaceCall(null, null, null, null, null, 5, null, null, Boolean.TRUE, null, null),
-                new TypeToken<Watch.Response<V1Namespace>>(){}.getType());
-
-        for (Watch.Response<V1Namespace> item : watch) {
-            System.out.printf("%s : %s%n", item.type, item.object.getMetadata().getName());
-        }
-    }
-}
-```
+We prepared a few examples for common user-cases as is shown below:
```
Comment (brendandburns) [inline README.md]: ... common use-cases which are shown below:`

File: README.md [hunk 2]
```diff
@@ -61,70 +61,46 @@ Then manually install the following JARs:
 
 ## Example
 
-list all pods:
-
-```java
```
Comment (brendandburns) [inline README.md]: Let's leave the simple code example in also, in addition to the list of examples you added.

File: examples/src/main/java/io/kubernetes/client/examples/InClusterClientExample.java [hunk 3]
```diff
@@ -1,15 +1,3 @@
-/*
```
Comment (brendandburns) [inline examples/src/main/java/io/kubernetes/client/examples/InClusterClientExample.java]: Please retain copyright header.

File: examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/** Alipay.com Inc. Copyright (c) 2004-2019 All Rights Reserved. */
```
Comment (brendandburns) [inline examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java]: Remove and add the Kubernetes copyright header.

File: examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * A simple example of how to use the Java API from an application outside a kubernetes cluster
+ *
+ * <p>Easiest way to run this: mvn exec:java
+ * -Dexec.mainClass="io.kubernetes.client.examples.OutOfClusterClientExample"
+ *
+ * <p>From inside $REPO_DIR/examples
+ */
+public class OutOfClusterClientExample {
```
Comment (brendandburns) [inline examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java]: Perhaps `ConfigFileClusterExample` is a better name? Most people should just use the 'Config.defaultClient()' constructor which should work both inside a cluster and outside a cluster.

**Comment by brendandburns** [comment]
Few small comments.

File: README.md (hunk 2 above)
Comment (yue9944882) [reply README.md]: i kept the simple example and added that to the list. wdyt?

File: examples/src/main/java/io/kubernetes/client/examples/InClusterClientExample.java (hunk 3 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/InClusterClientExample.java]: fixed

File: examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java (hunk 5 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java]: yes, considering kubeconfig can be used either in cluster or out. renamed to `KubeConfigFileClientExample`?

File: examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java (hunk 4 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/OutOfClusterClientExample.java]: oh dang, my IDE tooling always leaves me a surprise..

File: README.md (hunk 2 above)
Comment (brendandburns) [reply README.md]: I feel like if I maven import this library, having to do an extra click to get an example is too much work. I'd still like to include both.

**Comment by brendandburns** [comment]
This LGTM, modulo the comment on README.md

**Pull request merged**